ui/audio_language_selection.js

```diff
@@ -89,7 +90,12 @@ shaka.ui.AudioLanguageSelection = class extends shaka.ui.SettingsMenu {
    * @private
    */
   onAudioTrackSelected_(track) {
-    this.player.selectAudioLanguage(track.language, track.roles[0]);
+    if (track.channelsCount) {
```

Comment:
This should probably also check for `showAudioChannelCountVariants` being true.
Otherwise, when `showAudioChannelCountVariants` is false, I think this will result in the player selecting whichever track for the given language+role is listed first, right, since that track will be the one used to make the button? Instead of freely picking between different channel counts for the given language+role.

Reply:
Fixed!